manifests/reposync-crd.yaml

```diff
@@ -1186,8 +1186,52 @@ spec:
                     type: object
                   values:
                     description: values to use instead of default values that accompany
-                      the chart
+                      the chart. Format values the same as default values.yaml. These
+                      values will take precedence if used in conjunction with valuesFrom.
+                      How to handle multiple valuesFiles is determined by `valuesKeyMergeMode`.
                     x-kubernetes-preserve-unknown-fields: true
+                  valuesFrom:
+                    description: valuesFrom holds references to objects in the cluster
+                      that represent values to use instead of default values that
+                      accompany the chart. Currently, only ConfigMaps are supported.
+                      Objects listed later will take precedence. How to handle multiple
+                      valuesFiles is determined by `valuesKeyMergeMode`.
+                    items:
+                      description: ValuesFrom holds references to objects in the cluster
+                        that represent values to use instead of default values that
+                        accompany the chart.
+                      properties:
+                        key:
+                          default: values.yaml
+                          description: key represents the object data key to read
+                            the value from.
+                          type: string
+                        kind:
+                          description: kind represents the Object kind. Must be `ConfigMap`.
+                            Required.
+                          enum:
+                          - ConfigMap
+                          type: string
+                        name:
+                          description: name represents the Object name. Required.
+                          type: string
+                      type: object
+                    type: array
+                  valuesKeyMergeMode:
+                    default: override
+                    description: valuesKeyMergeMode specifies the strategy for handling
+                      multiple valueFiles. It refers to how values are merged together
+                      if multiple valuesFile define the same key. Can be 'override'
+                      or 'merge'. 'override' (default) results in the duplicated keys
+                      in later files to override the value from earlier files. This
+                      is equivalent to passing in multiple valuesFiles to Helm CLI.
+                      'merge' results in the duplicated keys being merged together
```

Comment:
This description of merging could be more clear... 
Is it just strings that are merged? how are lists and maps merged?
Is this a helm feature or a CS-specific feature?
Did a customer ask for it?

It feels like "merge" is a less than ideal name if not all field types are merged...

Reply:
This is a requirement for fleet tenancy integration. I can update the description more clearly. Lists are concatenated, maps are merged, scalars are overridden (since there's no real way to "merge" those).

Reply:
Also echoing my above comment, we had a really, really long discussion on this name and I'd prefer not to open that up again